draft-mouris-cfrg-mastic.md

```diff
@@ -661,15 +680,154 @@ malicious Aggregator.
+err that represents an error (e.g., `dns.unreachable`) or an indication that no
+error occurred (e.g., `ok`). Notably, this can be easily extended in Mastic to
+represent more elaborate metrics. e.g., where each weight includes the time it
+took each browser to report the error (and the aggregate is the average error
+reporting time), user agent (browser type and version), etc. However, our main
+goal is to understand 1) the distribution of errors and 2) which domains are
+impacted.
+
+We expect there to be a large number of distinct domain names (millions in the
+case of content delivery networks) and only a small number of error variants
+(the NEL spec {{W3C23}} defines 30 variants). The following Mastic parameters
+are suitable for this application.
+
+* **Inputs:** Each input `alpha` encodes the domain `dom` truncated to `n = 256`
+bits, which is sufficient to represent most of the domains on the internet
+{{MST24}}, {{BBCGGI21}}. Domains that are shorter than `n` bits are padded with
+0s.
+* **Weights:** Each weight `beta` represents the error variant `dom`. To compute
+the distribution of errors, we encode each error variant as a distinct bucket of
+a histogram so that `[1, 0, 0, ...]` represents `ok`, `[0, 1, 0, ...]`
+represents `dns.unreachable`, `[0, 0, 1, ...]` represents
+`dns.name_not_resolved`, and so on. There are 30 such variants (see {{W3C23}},
+Section 6), so the language `L` of valid weights is exactly the set of length-30
+vectors over `F` containing all 0s except for a single 1.
+* **Ordering:** Our `order(·)` function computes the ratio of reports with `err
+!= ok` to reports with `err = ok`. The latter is simply the first bucket of the
+aggregated histogram; the former is the sum of the remaining 29 buckets. Note
+that our ordering of aggregated weights considers the error rate rather than the
+raw error count. This ensures that the signal for less popular domains is not
+swamped by the noise generated by popular sites (network issues may impact some
+domains but not others). Another benefit is that, under normal operating
+conditions, there will be a small number of heavy-hitting domains, which means
+Mastic will run very efficiently. During an incident, there will be more heavy
+hitters, which means it will take longer to compute the set of impacted domains.
+However, we get the errors immediately at the root of the prefix tree, which is
+the most important information needed to begin remediation. As more levels are
+evaluated we get more detailed errors.
+
+
+## Attribute-Based Browser Telemetry {#attribute-based-telemetry}
+Web browsers, like Chrome, Firefox, or Safari, collect telemetry generated by
+users as they surf the web to gain insights into trends that guide product
+decisions. In many cases, Prio can be used to privately aggregate this
+telemetry. However, this comes at the cost of flexibility.
+
+For example, Mozilla is using Prio to collect page load metrics from Firefox for
```

Comment:
We'll need to remove mention Mozilla by name and speak about a hypothetical browser.